### PR TITLE
Add crds for continuous benchmark tools

### PR DESCRIPTION
**Pull request by vankichi**
Signed-off-by: vankichi <kyukawa315@gmail.com>

### Description:

I have created the crd and rbac for the continuous benchmark operator.

### Environment:

- Go Version: 1.18.3
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 1.14.7

### Types of changes:

- [ ] Bug fix [type/bug]
- [x] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [x] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by kpango — CHANGES_REQUESTED** [review]
Sorry I couldn't find struct which implements `k8s.io/apimachinery/pkg/runtime.Object`

File: internal/k8s/benchmark/operator/operator.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		opt(r)
+	}
+	return r
+}
+
+func (r *reconciler) AddListOpts(opt client.ListOption) {}
+
+//+kubebuilder:rbac:groups=vald.vdaas.org.vald.vdaas.org,resources=valdbenchmarkoperators,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=vald.vdaas.org.vald.vdaas.org,resources=valdbenchmarkoperators/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=vald.vdaas.org.vald.vdaas.org,resources=valdbenchmarkoperators/finalizers,verbs=update
```
Comment (kpango) [inline internal/k8s/benchmark/operator/operator.go:54]:
```suggestion
```

File: internal/k8s/benchmark/job/job.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		opt(r)
+	}
+	return r
+}
+
+func (r *reconciler) AddListOpts(opt client.ListOption) {}
+
+//+kubebuilder:rbac:groups=vald.vdaas.org.vald.vdaas.org,resources=valdbenchmarkjobs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=vald.vdaas.org.vald.vdaas.org,resources=valdbenchmarkjobs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=vald.vdaas.org.vald.vdaas.org,resources=valdbenchmarkjobs/finalizers,verbs=update
```
Comment (kpango) [inline internal/k8s/benchmark/job/job.go:54]:
```suggestion
```

File: internal/k8s/benchmark/job/job.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	namespace   string
+	onError     func(err error)
+	onReconcile func(jobList map[string][]BenchmarkJobSpec)
+	lopts       []client.ListOption
+}
+
+func New(opts ...Option) BenchmarkJobWatcher {
+	r := new(reconciler)
+	for _, opt := range append(defaultOpts, opts...) {
+		opt(r)
```
Comment (hlts2) [inline internal/k8s/benchmark/job/job.go:45]: Could you please add error handling for applying option? 🙏

File: internal/k8s/benchmark/operator/operator.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	namespace   string
+	onError     func(err error)
+	onReconcile func(operatorList map[string][]BenchmarkOperatorSpec)
+	lopts       []client.ListOption
+}
+
+func New(opts ...Option) BenchmarkOperatorWatcher {
+	r := new(reconciler)
+	for _, opt := range append(defaultOpts, opts...) {
+		opt(r)
```
Comment (hlts2) [inline internal/k8s/benchmark/operator/operator.go:45]: Could you please add error handling for applying option? 🙏

File: internal/k8s/benchmark/operator/types.go [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+	Dataset *job.BenchmarkDataset
+	Jobs    []*job.BenchmarkJobSpec
+}
+
+type BenchmarkOperatorStatus string
+
+const (
+	BenchmarkOperatorNotReady  = BenchmarkOperatorStatus("NotReady")
+	BenchmarkOperatorAvailable = BenchmarkOperatorStatus("Available")
+	BenchmarkOperatorHealthy   = BenchmarkOperatorStatus("Healthy")
```
Comment (hlts2) [inline internal/k8s/benchmark/operator/types.go:36]:
```suggestion
	BenchmarkOperatorNotReady BenchmarkOperatorStatus = "NotReady"
	BenchmarkOperatorAvailable   BenchmarkOperatorStatus = "Available"
	BenchmarkOperatorHealthy     BenchmarkOperatorStatus = "Healthy"
```

**Review by kpango — APPROVED** [review]

**Review by hlts2 — APPROVED** [review]

**Review by kevindiu — APPROVED** [review]

**Pull request merged**